acestep/llm_inference.py

```
@@ -125,16 +125,32 @@ def _clear_accelerator_cache(self) -> None:
         """Release LM weights/tokenizer and clear caches to free memory."""
         try:
-            if self.llm_backend == "vllm":
+            if self.llm_backend == "vllm" and getattr(self, "llm", None) is not None:
+                exit_success = False
                 try:
-                    if hasattr(self.llm, "reset"):
-                        self.llm.reset()
+                    if hasattr(self.llm, "exit"):
+                        self.llm.exit()  # full teardown: weights/KV cache/CUDA graphs/subprocesses
+                        exit_success = True
+                except Exception as exc:
```

Ruff (0.16.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)